app/controllers/application_controller.rb

```diff
@@ -21,6 +21,21 @@ def current_user
   end
   helper_method :current_user
 
+  def current_group
+    if params[:group_id]
+      @group ||= Group.find(params[:group_id])
+    else
+      @group ||= Group.find(params[:id])
+    end
+    @group
+  end
+  helper_method :current_group
+
+  def require_leader
+    membership = current_user.membership_for(current_group)
+    unauthorized_page unless membership && membership.leader?
```

Comment:
For now this is adequate although later you might need more permission handling. maybe something like pundit or a different authorization gem can help clear this logic out from the application controller.

also for syntax, what do you think about something like this? `membership.try!(:leader?)`. should be the same just more "rails way". but in the end it's a matter of your preference.

Reply:
We've checked pundit and we decided to delay its introduction, simply we already have a lot of code using this type of logic. Fortunately there are not too much authorization logic, so I hope we won't end up in a mess.

Could you specify a bit more how do you imagine this kind of syntax?